tests/test_users/test_routers.py

```diff
@@ -4,74 +4,76 @@
+from src.core.utils import datetime_encoder
 
 def test_create_user(
     sync_client: TestClient,
 ):
-    register_data = UserFactory.build(password="mtdqwc241", password_repeat="mtdqwc241")
-    response = sync_client.post("users/register", json=json.loads(register_data.json()))
+    register_data = UserRegisterSchemaFactory.build(password="mtdqwc241", password_repeat="mtdqwc241")
+    register_data.birth_date = datetime_encoder(register_data.birth_date)
+    response = sync_client.post("users/register", json=register_data.dict())
     assert response.status_code == status.HTTP_201_CREATED
-    
+
+
 def test_login_user(
     sync_client: TestClient,
-    login_data: dict[str, str],
     db_user: UserOutputSchema
 ):
+    login_data = {'username': DB_USER_SCHEMA.username, 'password': DB_USER_SCHEMA.password}
     response = sync_client.post("users/login", json=login_data)
     assert response.status_code == status.HTTP_200_OK
     assert "access_token" in response.json()
 
 def test_authenticated_user_can_get_users(
     sync_client: TestClient,
-    access_token: dict[str, str],
+    auth_headers: dict[str, str],
     db_user: UserOutputSchema
 ):
-    response = sync_client.get("users/", headers=access_token)
+    response = sync_client.get("users/", headers=auth_headers)
     
-    assert response.json()['total'] == len([db_user])
+    assert response.json()['total'] == 1
     assert response.status_code == status.HTTP_200_OK
     
 def test_authenticated_user_can_get_single_user(
     sync_client: TestClient,
-    access_token: dict[str, str],
+    auth_headers: dict[str, str],
     db_user: UserOutputSchema
 ):
-    response = sync_client.get(f"users/{db_user.id}", headers=access_token)
+    response = sync_client.get(f"users/{db_user.id}", headers=auth_headers)
     assert response.json()["id"] == db_user.id
     assert response.status_code == status.HTTP_200_OK
 
 
 def test_authenticated_user_can_get_their_account(
     sync_client: TestClient,
-    access_token: dict[str, str],
+    auth_headers: dict[str, str],
     db_user: UserOutputSchema
 ):
-    response = sync_client.get(f"users/me", headers=access_token)
+    response = sync_client.get(f"users/me", headers=auth_headers)
     assert response.json()["id"] == db_user.id
     assert response.status_code == status.HTTP_200_OK
 
 
 def test_authenticated_user_can_update_user(
     sync_client: TestClient,
-    access_token: dict[str, str],
+    auth_headers: dict[str, str],
     db_user: UserOutputSchema
 ):
     update_data = {"username": "alex123"}
-    response = sync_client.patch(f"users/{db_user.id}", json=update_data, headers=access_token)
-    print(response.json())
+    response = sync_client.patch(f"users/{db_user.id}", json=update_data, headers=auth_headers)
     assert response.status_code == status.HTTP_200_OK
-    assert response.json()["username"] == update_data["username"]
+    assert response.json()["username"] == update_data['username']

```

Comment:
Single quote wszedzie albo double wszedzie, bez mieszania. Najlepiej to formatery kodu sobie ogarnac (isort, black)